ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -165,9 +175,13 @@ void close() {
       // the RequestResponseHandler thread when it is in poll() or handleResponse(). In order to avoid the completion
       // from happening twice, complete it here only if the remove was successful.
       if (deleteOperations.remove(op)) {
+        RouterException routerException =
+            new RouterException("Aborted operation because Router is closed.", RouterErrorCode.RouterClosed);
         routerMetrics.operationDequeuingRate.mark();
-        operationCompleteCallback.completeOperation(op.getFutureResult(), op.getCallback(), null,
-            new RouterException("Aborted operation because Router is closed.", RouterErrorCode.RouterClosed));
+        routerMetrics.operationAbortCount.inc();
```

Comment:
this is the only place where you should update the `operationAbort` metric - when it is actually being aborted.